source/_components/insteon.markdown

```diff
@@ -81,129 +75,136 @@ insteon:
-  to the local Hub. This is required for Hub [2245] (i.e. Hub version 2) setup.
-  You can find your Hub password on the bottom of the Hub or you can use the
-  Insteon app.
-- **hub_version** (*Required for Hub version 1 setup*): The Hub version number
-  where model [2242] is Hub version 1 and model [2245] is Hub version 2.
-  (Default is 2)
-- **device_override** (*Optional*): Override the default device definition
-  - *ADDRESS* is found on the device itself in the form 1A.2B.3C or 1a2b3c
-  - *CATEGORY* is found in the back of the device's User Guide in the form of
-    0x00 - 0xff
-  - *SUBCATEGORY* is found in the back of the device's User Guide in the form 
-    of 0x00 - 0xff
-  - *FIRMWARE* and *PRODUCT_KEY* are more advanced options and will typically 
-    not be used.
-- **x10_devices** (*Optional*): Define X10 devices to control or respond to
-  - *HOUSECODE* is the X10 housecode values a - p
-  - *UNITCODE* is the X10 unit code values 1 - 16
-  - *PLATFORM* is the Home Assistant Platform to associate the device with. 
-    The following platforms are supported
-    - binary_sensor: Used for on/off devices or keypad buttons that are read-only.
-    - light: Used for dimmable X10 devices
-    - switch: Used for On/Off X10 devices
-  - *STEPS* is the number of dim/bright steps the device supports. Used for
-    dimmable X10 devices only. Default value is 22.
-- **x10_all_units_off** (*Optional*): Creates a binary_sensor that responds
-  to the X10 standard command for All Units Off.
-- **x10_all_lights_on** (*Optional*):  Creates a binary_sensor that responds
-  to the X10 standard command for All Lights On
-- **x10_all_lights_off** (*Optional*): Creates a binary_sensor that responds
-  to the X10 standard command for All Lights Off
+
+{% configuration %}
+port:
+  description: The serial or USB port for your device, e.g., `/dev/ttyUSB0` or `COM3`.
+  required: Required for PLM setup
+  type: string
+host:
+  description: The host name or IP address of the Hub.
+  required: Required for Hub setup
+  type: string
+ip_port:
+  description: The IP port number of the Hub. For Hub model [2245] (i.e. Hub version 2) the default port is 25105. For the Hub model [2242] (i.e. Hub version 1) the default port is 9761. Use the Insteon app to find the port number for your specific Hub.
+  required: Optional for Hub setup
+  type: integer
+username:
+  description: The username to login in to the local Hub. You can find your Hub username on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
+  type: string
+password:
+  description: The password to login in to the local Hub. You can find your Hub password on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
+  type: string
+hub_version:
+  description: The Hub version number where model [2242] is Hub version 1 and model [2245] is Hub version 2.
+  required: Required for Hub version 1 setup
```

Comment:
`required` should be a `true`/`false`/`inclusive`/`exclusive`.

Reply:
why does it say in the documentation that you can put a string? [Info](https://developers.home-assistant.io/docs/en/documentation_create_page.html)

Reply:
So why am I the one that didn't know? 😭 

Sorry about that m8!

Reply:
Well everyone learns something at any time 😉